packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1047,28 +1047,36 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
     if (_enabled) {
       items = widget.selectedItemBuilder == null
         ? List<Widget>.from(widget.items)
-        : widget.selectedItemBuilder(context).map<Widget>((Widget item) {
-            return Container(
-              constraints: const BoxConstraints(minHeight: _kMenuItemHeight),
-              alignment: AlignmentDirectional.centerStart,
-              child: item,
-            );
-          }).toList();
+        : widget.selectedItemBuilder(context);
     } else {
-      items = <Widget>[];
+      items = widget.selectedItemBuilder == null
+        ? <Widget>[]
+        : widget.selectedItemBuilder(context);
     }
 
     int hintIndex;
     if (widget.hint != null || (!_enabled && widget.disabledHint != null)) {
-      final Widget emplacedHint = _enabled
-        ? widget.hint
-        : DropdownMenuItem<Widget>(child: widget.disabledHint ?? widget.hint);
+      Widget displayedHint;
+      if (_enabled) {
+        displayedHint = widget.hint;
+      } else {
+        displayedHint = widget.disabledHint ?? widget.hint;
+      }
+
+      if (widget.selectedItemBuilder == null) {
+        displayedHint = Container(
```

Comment:
This `Container` mimics the constraints and alignments for a default menu item, similar to `DropdownMenuItem`

Reply:
It would probably be safest to either include a comment here and in DropdownMenuItem.build() indicating that the code must be kept in sync, or add a _DropdownMenuItemContainer widget and use it in both places. The widget approach might be preferable, since you can explain what's going on vis-a-vis selectedItemBuilder in a comment above the widget class.

Reply:
Sounds good, I just added a `_DropdownMenuItemContainer` and had `DropdownMenuItem` extend it to keep the code in sync. PTAL